webrtc/RTCDataChannel-binaryType.html

```diff
@@ -0,0 +1,69 @@
+<!doctype html>
+<meta charset=utf-8>
+<title>RTCDataChannel.prototype.binaryType</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="RTCPeerConnection-helper.js"></script>
+<script>
+'use strict';
+
+// Disable global timeout
+// IMPORTANT: You need to add a timeout to *every test* of this file!
+setup({ explicit_timeout: true });
```

Comment:
Removed (in this file).

Reply:
However, in general this exists to allow a per-test timeout instead of the inflexible global per-file timeout which can be either 10 seconds or 60 seconds. This is insufficient for a test file with many promise-based test cases that can only run sequentially. Therefore, these files provide a timeout for each test case and disable the global timeout.

Reply:
> If we fear that timing out subtests is an issue, we should simply split subtests in different files.

Don't forget that a `promise_test` blocks all subsequent tests. This means, if a `promise_test` does not have a local timeout and it does time out, it prevents all other tests from running. If we don't want local timeouts, then we have to move each test case into its own file.

Reply:
In practice, timing out subtests do happen.
Browser implementors tend to try fixing the corresponding issues sooner than regular failures.

Reply:
I very much like the thought of browser vendors fixing their implementations due to tests timing out but don't consider that approach a good thing. Anyhow, I have moved this discussion further down into the comments since I want to hear some more opinions on that.